notebooks/src/el-paso.ipynb

```diff
@@ -186,21 +190,47 @@
     "  if bw is not None:\n",
     "      q.show_prediction(samples=q_samples, show_community=True, percent_kept=0.9, bw=bw)\n",
     "  else:\n",
-    "      q.show_prediction(samples=q_samples, show_community=True, percent_kept=0.9)"
+    "      q.show_prediction(samples=q_samples, show_community=True, percent_kept=0.9)\n",
+    "\n",
+    "\n",
+    "def sample_from_ensemble(models, params, weights=None, fallback=False, default=None):\n",
+    "    \"\"\"Sample models in proportion to weights and execute with\n",
+    "    model_params. If fallback is true then call different model from\n",
+    "    ensemble if the selected model throws an error. If Default is not\n",
+    "    None then return default if all models fail\n",
+    "\n",
+    "    \"\"\"\n",
+    "\n",
+    "    model = ergo.random_choice(models, weights)\n",
+    "    try:\n",
+    "        result = model(**params)\n",
+    "        if np.isnan(result):\n",
+    "            raise KeyError\n",
+    "        return result\n",
+    "    except:\n",
+    "        if fallback and len(models) > 1:\n",
+    "            i = models.index(model)\n",
+    "            del models[i]\n",
```

Comment:
This mutates the arguments to `sample_from_ensemble`, could lead to unintended side-effects.

Reply:
@stuhlmueller So I should make copies and pass them in instead of mutating the original arguments?

Reply:
Yes